modules/atomic-widgets/prop-types/base/array-prop-type.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Elementor\Modules\AtomicWidgets\PropTypes\Base;
+
+use Elementor\Modules\AtomicWidgets\PropTypes\Concerns;
+use Elementor\Modules\AtomicWidgets\PropTypes\Contracts\Transformable_Prop_Type;
+use Elementor\Modules\AtomicWidgets\PropTypes\Contracts\Prop_Type;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+abstract class Array_Prop_Type implements Transformable_Prop_Type {
+	const TYPE = 'array';
+
+	use Concerns\Has_Meta,
+		Concerns\Has_Settings,
+		Concerns\Has_Default,
+		Concerns\Has_Transformable_Validation,
+		Concerns\Has_Required_Validation;
+
+	protected Prop_Type $item_type;
+
+	public function __construct() {
+		$this->item_type = $this->define_item_type();
+	}
+
+	public static function make(): self {
+		return new static();
+	}
+
+	public function set_item_type( Prop_Type $item_type ): self {
+		$this->item_type = $item_type;
+
+		return $this;
+	}
+
+	public function get_item_type(): Prop_Type {
+		return $this->item_type;
+	}
+
+	public function default( $value ): self {
+		$this->default = [
+			'$$type' => static::get_key(),
+			'value' => $value,
+		];
+
+		return $this;
+	}
+
+	public function validate( $value ): bool {
+		if ( is_null( $value ) ) {
+			return ! $this->is_required();
+		}
```

Comment:
maybe extract to a trait?

Reply:
I did not find any way to do so... I'm not sure how.

Reply:
maybe:
```
public function validate_required( $value ) {
    if ( is_null( $value ) ) {
	return ! $this->is_required();
    }
    
    return true;
}